### PR TITLE
Dikshit/open wire check v2

### PR DESCRIPTION
**Pull request by DJ90864**
### Summary

Better task management for between OWC and normal cell measurements

### Testing Done

tested on thruna BMS core, seems to be working. Still needs to be tested on Quadruna

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: firmware/quadruna/BMS/src/app/states/app_allStates.c [hunk 1]
```diff
@@ -13,6 +13,16 @@
 // Num of cycles for voltage and cell temperature values to settle
 #define NUM_CYCLES_TO_SETTLE (30U)
 
+uint32_t owcCounter = 0;
```
Comment (will-chaba) [inline firmware/quadruna/BMS/src/app/states/app_allStates.c:16]: Variables typically in snake_case
```suggestion
uint32_t iso_spi_state_counter = 0;
```

File: firmware/quadruna/BMS/src/app/states/app_allStates.c [hunk 2]
```diff
@@ -13,6 +13,16 @@
...
+
+typedef enum
+{
+    RUN_CELL_MEASUREMENTS,
+    RUN_CELL_OWC
+} isoSpiTasks;
+
+isoSpiTasks isoSpiTasksState = RUN_CELL_MEASUREMENTS;
```
Comment (will-chaba) [inline firmware/quadruna/BMS/src/app/states/app_allStates.c:24]:
```suggestion
isoSpiTasks iso_spi_task_state = RUN_CELL_MEASUREMENTS;
```

File: firmware/thruna/BMS/src/app/app_accumulator.c [hunk 3]
```diff
@@ -15,6 +15,7 @@
 
 // Number of open wire check commands (ADOW) to send before open wire check
 #define OPEN_WIRE_CHECK_NUM_ADOW_CMDS (2)
+#define OPEN_WIRE_CHCEK_EMPTY_CYCLES (4)
```
Comment (will-chaba) [inline firmware/thruna/BMS/src/app/app_accumulator.c:18]: nit: 
```suggestion
#define OPEN_WIRE_CHCEK_IDLE_CYCLES (4)
```

File: firmware/thruna/BMS/src/app/app_accumulator.c [hunk 4]
```diff
@@ -62,6 +63,7 @@ typedef enum
     GET_PU_CELL_VOLTAGE_STATE,
     GET_PD_CELL_VOLTAGE_STATE,
     CHECK_OPEN_WIRE_FAULT_STATE,
+    EMPTY_STATE,
```
Comment (will-chaba) [inline firmware/thruna/BMS/src/app/app_accumulator.c:66]: nit: 
```suggestion
    IDLE_STATE,
```

File: firmware/thruna/BMS/src/app/app_accumulator.c [hunk 5]
```diff
@@ -106,6 +108,7 @@ typedef struct
 static Accumulator data;
 static uint8_t     open_wire_pu_readings;
 static uint8_t     open_wire_pd_readings;
+static uint8_t     owcEmptyStates;
```
Comment (will-chaba) [inline firmware/thruna/BMS/src/app/app_accumulator.c:111]:
```suggestion
static uint8_t     owc_idle_cycles;
```

File: firmware/thruna/BMS/src/app/states/app_allStates.c [hunk 6]
```diff
@@ -13,6 +13,16 @@
 // Num of cycles for voltage and cell temperature values to settle
 #define NUM_CYCLES_TO_SETTLE (30U)
 
+uint32_t owcCounter = 0;
+
+typedef enum
+{
+    RUN_CELL_MEASUREMENTS,
+    RUN_CELL_OWC
```
Comment (will-chaba) [inline firmware/thruna/BMS/src/app/states/app_allStates.c:21]: To stay consistent with function name
```suggestion
    RUN_OPEN_WIRE_CHECK
```

**Review by will-chaba — CHANGES_REQUESTED** [review]
Looks great! just some naming change suggestions and switch variables to snake_case

**Comment by DJ90864** [comment]
@will-chaba requested changes done!

**Pull request merged**